File: exercises/D106/solution.cpp
#include <iostream>
using namespace std;
int main() {
    int n;
    cin >> n;
    int units = n % 10;
    cout << n;
    if (n >= 10 && n / 10 % 10 == 1) {
        cout << "th" << endl;
    } else if (units == 1) {
        cout << "st" << endl;
    } else if (units == 2) {
        cout << "nd" << endl;
    } else if (units == 3) {
        cout << "rd" << endl;
    } else {
        cout << "th" << endl;
    }
}

File: exercises/D106/solution.py
n = int(input())
units = n % 10
if n >= 10 and n // 10 % 10 == 1:
    print(f'{n}th')
elif units == 1:
    print(f'{n}st')
elif units == 2:
    print(f'{n}nd')
elif units == 3:
    print(f'{n}rd')
else:
    print(f'{n}th')
